tests/e2e/attention/generate_e2e_fa2_tests.py

```diff
@@ -0,0 +1,516 @@
+    op = op + generate_random_3d_tensor("key", key_shape, key_type)
+    op = op + generate_random_3d_tensor("value", value_shape, value_type)
+
+    global pseudorandom_generator_seed
+    pseudorandom_generator_seed = pseudorandom_generator_seed - 1
+    op = op + (
+        f"  %scale = arith.constant {shapes_scale.scale} : f32\n"
+        f"  %result = call @module.{function.name}(%query, %key, %value, %scale) : (!hal.buffer_view, !hal.buffer_view, !hal.buffer_view, f32) -> !hal.buffer_view\n"
+    )
+
+    op = op + (
+        f"  %batch = arith.constant {shapes_scale.batch} : i64\n"
+        f"  %m = arith.constant {shapes_scale.m} : i64\n"
+        f"  %k1 = arith.constant {shapes_scale.k1} : i64\n"
+        f"  %k2 = arith.constant {shapes_scale.k2} : i64\n"
+        f"  %n = arith.constant {shapes_scale.n} : i64\n"
+        f"  call @fa2_test.check_fa2_results(%device, %batch, %m, %k1, %k2, %n, %query, %key, %value, %result) : (!hal.device, i64, i64, i64, i64, i64, !hal.buffer_view, !hal.buffer_view, !hal.buffer_view, !hal.buffer_view) -> ()\n"
+    )
+
+    op = op + "  return\n"
+    op = op + "}\n"
+
+    return TestCall(function=function, op=op)
+
+
+# Generates all output files' contents as strings.
+def generate(
+    query_type: QueryElemTypeId,
+    key_type: KeyElemTypeId,
+    value_type: ValueElemTypeId,
+    shapes_id: ShapesId,
+):
+    functions = {}
+    calls = []
+
+    for shape in get_test_shapes(shapes_id):
+        function = generate_function(
+            query_type,
+            key_type,
+            value_type,
+            shape,
+        )
+        if function.name not in functions:
+            functions[function.name] = function
+        calls.append(
+            generate_call(
+                function,
+                query_type,
+                key_type,
+                value_type,
+                shape,
+            )
+        )
+
+    return (functions, calls)
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description="Generator of e2e fa2 tests")
+    parser.add_argument(
+        "--output_fa2_mlir",
+        type=str,
+        help="Path of output .mlir file containing the generated fa2 functions",
+        required=True,
+    )
+    parser.add_argument(
+        "--output_calls_mlir",
+        type=str,
+        help="Path of output .mlir file containing the calls",
+        required=True,
+    )
+    parser.add_argument(
+        "--query_type",
+        type=str,
+        choices=["f16"],
+        help="Numeric type of query tensors ",
+        required=True,
+    )
+    parser.add_argument(
+        "--key_type",
+        type=str,
+        choices=["f16"],
+        help="Numeric type of key tensors ",
+        required=True,
+    )
+    parser.add_argument(
+        "--value_type",
+        type=str,
+        choices=["f16"],
+        help="Numeric type of value tensors ",
+        required=True,
+    )
+    parser.add_argument(
+        "--shapes_scale",
```

Comment:
It can just be a `--shapes` flag.

Reply:
I arranged it so we can add a test case with different scaling.
```
    batch = shapes_scale.batch
    m = shapes_scale.m
    k1 = shapes_scale.k1
    k2 = shapes_scale.k2
    n = shapes_scale.n
    scale = shapes_scale.scale
```